packages/legend-query-builder/src/components/result/tds/QueryBuilderTDSSimpleGridResult.tsx

```diff
@@ -47,6 +48,87 @@ import type {
 } from '../../../stores/QueryBuilderResultState.js';
 import { QUERY_BUILDER_TEST_ID } from '../../../__lib__/QueryBuilderTesting.js';
 
+export const getFloatGridColumnCustomHeader = (columnName: string): string =>
```

Comment:
Is it not possible for this function to return a `React.ReactNode` or `JSX.Element`? Seems a big fragile having it return a string.

Reply:
Also, this seems to be duplicating a lot of HTML that's probably built-in to ag-grid. Is there no way to just append a custom element (the warning tooltip) to the default header, rather than recreate all these `span` and `div` elements?

Reply:
I'm currently only able to not only add a warning icon but also keep built-in ag-grid header functionalities using this approach. reference: https://www.ag-grid.com/react-data-grid/column-headers/#provided-component

I have tried to use 
 {
        headerComponent: ${},
        headerComponentParams: {
          displayName: ${},
        },
 };


https://github.com/finos/legend-studio/commit/f042dde23a77ad6732aabc831c4748fbdc6aa352#diff-45e34618f522207e317ad27fff3adf3289a41ce503d68b2f2b4b7c0b27982aecR53

But I currently failed to keep built-in ag-grid header functionalities